src/components/meta-data/index.jsx

```diff
@@ -0,0 +1,496 @@
+				...accumulator,
+				[key]: [
+					...(typeof accumulator[key] === "object" ? accumulator[key] : [accumulator[key]]),
+					value,
+				],
+			};
+		}
+		return { ...accumulator, [key]: value };
+	}, {});
+};
+
+const getPageCanonicalUrl = (pageType, domain, globalContent, requestUri) => {
+	const urlParameters = getUrlParameters(requestUri);
+	const authorCanonicalPath = globalContent?.authors ? globalContent?.authors[0]?.bio_page : "";
+	const globalCanonicalPath = globalContent?.canonical_url;
+	const querylyCanonicalPath = urlParameters.query
+		? `${requestUri.replace(/\?.*/, "")}?query=${urlParameters.query}`
+		: "";
+	const searchCanonicalPath = globalContent?.metadata ? `search/${globalContent?.metadata.q}/` : "";
+	const tagCanonicalPath = globalContent?.Payload ? `tags/${globalContent?.Payload[0]?.slug}/` : "";
+
+	const canonicalUrlMapping = {
+		article: buildUrl(domain, globalCanonicalPath),
+		author: buildUrl(domain, authorCanonicalPath),
+		gallery: buildUrl(domain, globalCanonicalPath),
+		homepage: domain,
+		"queryly-search": buildUrl(domain, querylyCanonicalPath),
+		search: buildUrl(domain, searchCanonicalPath), // arc search
+		section: buildUrl(domain, globalContent?._id || ""),
+		tag: buildUrl(domain, tagCanonicalPath),
+		video: buildUrl(domain, globalCanonicalPath),
+	};
+	return canonicalUrlMapping[pageType];
+};
+
+const generateUrl = (arcSite, websiteDomain, gc) => {
+	const siteData = gc && gc.websites && gc.websites[arcSite];
+	if (!siteData) {
+		return null;
+	}
+	return `${websiteDomain}${siteData.website_url}`;
+};
+
+const normalizeFallbackImage = (websiteDomain, url) => {
+	if (!url) {
+		return null;
+	}
+	if (!url.startsWith("http")) {
+		const tmp = `${websiteDomain}¬${url}`;
+		return tmp.replace(/\/?¬\/?/, "/");
+	}
+	return url;
+};
+
+const MetaData = ({
+	arcSite,
+	canonicalDomain,
+	canonicalResolver,
+	facebookAdmins,
+	fallbackImage,
+	globalContent: gc,
+	MetaTag,
+	MetaTags,
+	metaValue,
+	outputCanonicalLink,
+	requestUri,
+	resizerURL,
+	twitterUsername,
+	websiteDomain,
+	websiteName,
+}) => {
+	const pageType = metaValue("page-type");
+
+	let storyMetaDataTags = null;
+	let tagMetaDataTags = null;
+	let authorMetaDataTags = null;
+	let searchMetaDataTags = null;
+	let sectionMetaDataTags = null;
+	let homepageMetaDataTags = null;
+	let nativoMetaDataTags = null;
+	let commonTagsOnPage = true;
+	let canonicalLink = null;
+
+	const metaData = {
+		description: null,
+		keywords: null,
+		ogImage: null,
+		ogImageAlt: null,
+		url: generateUrl(arcSite, websiteDomain, gc),
+		"page-type": pageType,
+		title: metaValue("title") || websiteName,
+		"og:title": metaValue("og:title") || websiteName,
+		ogSiteName: websiteName,
+		twitterUsername: twitterUsername ? `@${twitterUsername}` : null,
+		twitterCard: "summary_large_image",
+		"twitter:title": metaValue("twitter:title") || websiteName,
+		twitterImage: null,
+		fallbackImage: normalizeFallbackImage(websiteDomain, fallbackImage),
```

Comment:
added resizing for the fallback, but I have some questions about how it should be resized. I have it set to width 1200 and no height currently. since it's set to smart, it'll automatically determine the height based on the aspect ratio. I can set it to be 630 like OG images, but I noticed that our fallback is long and skinny (I don't know if this is typical), so it crops it weird. Here's the difference between no height and 630px:

https://themesinternal-themesinternal-sandbox.web.arc-cdn.net/resizer/v2/https%3A%2F%2Fstatic.themebuilder.aws.arc.pub%2Fthemesinternal-custom1-sandbox%2F1636037887909.png?smart=true&auth=91e862279e2e21d94ed7539546a8207ba98ac1763a7f9559c9efd29c4bd487a3&width=1200

https://themesinternal-themesinternal-sandbox.web.arc-cdn.net/resizer/v2/https%3A%2F%2Fstatic.themebuilder.aws.arc.pub%2Fthemesinternal-custom1-sandbox%2F1636037887909.png?smart=true&auth=91e862279e2e21d94ed7539546a8207ba98ac1763a7f9559c9efd29c4bd487a3&width=1200&height=630

Reply:
Having it resize with a width and no height set seems correct to me. Since we won't know the aspect ratio, and the fallback image is typically a logo, we don't want any cropping to occur.